obspy/signal/interpolation.py

```diff
@@ -172,6 +173,255 @@ def weighted_average_slopes(data, old_start, old_dt, new_start, new_dt,
+    :math:`t_j`:
+
+    .. math::
+
+        \begin{align}
+            S(t_j) = \sum_{i = \left \lfloor{x}\right \rfloor - a + 1}
+                          ^{\left \lfloor{x}\right \rfloor + a}
+            s_i L(t_j - i),
+        \end{align}
+
+    where :math:`\lfloor \cdot \rfloor` denotes the floor function. For more
+    details and justification please see [Burger2009]_ and [vanDriel2015]_.
+    """
+    _validate_parameters(data, old_start, old_dt, new_start, new_dt, new_npts)
+    dt_factor = float(new_dt) / old_dt
+    offset = new_start - old_start
+    if offset < 0:
+        raise ValueError("Cannot extrapolate.")
+
+    if a < 1:
+        raise ValueError("a must be at least 1.")
+
+    return_data = np.zeros(new_npts, dtype="float64")
+
+    clibsignal.lanczos_resample(
+        np.require(data, dtype=np.float64), return_data, dt_factor, offset,
+        len(data), len(return_data), int(a), 0)
+    return return_data
+
+
+def calculate_lanczos_kernel(x, a, window):
+    """
+    Helper function to get the actually used kernel for a specific value of a.
+
+    :type x: :class:`numpy.ndarray`
+    :param x: The x values at which to calculate the kernel.
+    :type a: int
+    :param a: The width of the window in samples on either side.
+    :type window: str
+    :param window: The window used to multiply the sinc function with. One
+        of ``"lanczos"``, ``"hanning"``, ``"blackman"``.
+
+    Return a dictionary of arrays.
+    """
+    window = window.lower()
+    if window not in _LANCZOS_KERNEL_MAP:
+        msg = "Invalid window. Valid windows: %s" % ", ".join(
+            sorted(_LANCZOS_KERNEL_MAP.keys()))
+        raise ValueError(msg)
+
+    x = np.require(x, dtype=np.float64)
+    y0 = np.zeros(x.shape, dtype=np.float64)
+    y1 = np.zeros(x.shape, dtype=np.float64)
+    y2 = np.zeros(x.shape, dtype=np.float64)
+
+    clibsignal.calculate_kernel(
+        x, y0, len(x), a, 0, _LANCZOS_KERNEL_MAP[window])
+    clibsignal.calculate_kernel(
+        x, y1, len(x), a, 1, _LANCZOS_KERNEL_MAP[window])
+    clibsignal.calculate_kernel(
+        x, y2, len(x), a, 2, _LANCZOS_KERNEL_MAP[window])
+
+    ret_val = {
+        "full_kernel": y0,
+        "only_sinc": y1,
+        "only_taper": y2
+    }
+
+    return ret_val
+
+
+def plot_lanczos_windows(a):
+    """
+    Helper function producing a plot of all available tapers of the sinc
+    function and their response for the Lanczos interpolation.
+
+    :type a: int
+    :param a: The width of the window in samples on either side.
+
+    .. code-block:: python
+
+        from obspy.signal.interpolation import plot_lanczos_windows
+        plot_lanczos_windows(a=20)
+
+    .. plot::
+
+        import matplotlib.pyplot as plt
+        plt.figure(figsize=(10, 15))
+        from obspy.signal.interpolation import plot_lanczos_windows
+        plot_lanczos_windows(a=20)
+    """
+    x_max = 1024.0 - 0.5
+    n = 2 ** 15
+    x = np.linspace(-x_max, x_max, n)
```

Comment:
Ah, yes, I guess I just saw `rfftfreq` in a loop and had that in the back of my mind. Maybe the call can still be moved out of the loop though?